bpm/bonita-api/bonita-common-api-impl/src/main/java/org/bonitasoft/engine/bpm/process/impl/ProcessDefinitionBuilder.java

```diff
@@ -126,6 +127,55 @@ private void validateProcess() {
         validateEventsSubProcess();
         validateActors();
         validateBusinessData();
+        validateProcessContract();
+    }
+
+    private void validateUserTask(final UserTaskDefinition userTaskDefinition) {
+        validateContract(userTaskDefinition.getContract(), "user task <" + userTaskDefinition.getName() + ">");
+    }
+
+    private void validateContractInputName(final String name) {
+        if (!SourceVersion.isIdentifier(name)) {
+            designErrors.add("contract input name " + name + " is invalid");
+        }
+    }
+
+    void validateProcessContract() {
+        final ContractDefinition contract = process.getContract();
+        validateContract(contract, "the process");
+    }
+
+    void validateContract(ContractDefinition contract, String containerIdentifier) {
+        if (contract == null) {
+            return;
+        }
+        for (final ConstraintDefinition constraint : contract.getConstraints()) {
+            if (constraint.getName() == null) {
+                addError("A constraint name is missing");
+            }
+            if (constraint.getExpression() == null) {
+                addError("The expression of constraint" + constraint.getName() + " is missing");
+            }
+        }
+        for (InputDefinition inputDefinition : contract.getInputs()) {
+            validateContractInput(containerIdentifier, inputDefinition);
+        }
+    }
+
+    private void validateContractInput(String containerIdentifier, InputDefinition inputDefinition) {
+        validateContractInputName(inputDefinition.getName());
+        if (inputDefinition.hasChildren() && (inputDefinition.getType() != null && !inputDefinition.getType().equals(Type.FILE))) {
+            addError("Can't have a type set on the contract input <" + inputDefinition.getName() + "> of contract of " + containerIdentifier
```

Comment:
it results in that: "Type not set on the contract input <name> of contract of user task <step1>"